spot-contracts/contracts/PerpetualTranche.sol

```diff
@@ -629,6 +635,11 @@ contract PerpetualTranche is
     ///      This function is to be invoked on all external function entry points which are
     ///      read the reserve storage. This function is intended to be idempotent.
     function updateState() public override {
+        // Skip state update when system is paused.
+        if (paused()) {
```

Comment:
Normally we ensure pause state with the `whenNotPaused` modifier. Any reason you deviated?

That modifier reverts, so is it because to keep execution going?

I think it's safer to revert, in case there are paths that keep going, for whatever reason, that expect an updated state but don't have one. That could lead to weird scenarios.

Reply:
We use `whenNotPaused afterStateUpdate` modifiers for deposit, redeem and rollover methods and the `afterStateUpdate` modifier for a bunch of lazy exec read methods. If the `updateState` method reverts, it will brick all of them. I think we still want those methods to work when paused.

Reply:
Agree, that's probably right. Now that the vault is the only way to roll over, pause is really for "sky is falling" scenarios so we should be mindful of that.